src/httpClient.ts

```diff
@@ -143,8 +146,8 @@ export default class HttpClient {
     data: any,
     config: AxiosRequestConfig = {},
   ): Promise<AxiosResponse<T>> {
-    config.headers = await this.createHeadersWithResolvedToken(config.headers);
-    return this.client.post<T>(url, data, config);
+    const headers = await this.createHeadersWithResolvedToken(config.headers);
+    return this.client.post<T>(url, data, { ...config, headers });
```

Comment:
This is unrelated - just to remove the eslint-disable. It has been done like this for GET but not the other methods https://github.com/Cimpress-MCP/AT-HttpClient/pull/2/files#diff-d41127e126779fcf600668763525c9d8ed29e25e39c31f9cfb3d2b0282791acaL133